src/components/lib/sl/sl_sched.c

```diff
@@ -260,6 +305,60 @@ sl_thd_block_periodic(thdid_t tid)
 	return jitter;
 }
 
+void
+sl_thd_block_expiry(struct sl_thd *t)
+{
+	cycles_t abs_timeout = 0;
+
+	assert(t != sl__globals()->sched_thd);
+	if (!(t->properties & SL_THD_PROPERTY_OWN_TCAP)) {
+		assert(!t->sched_blocked);
+		return;
+	}
+	assert(t->period);
+
+	sl_cs_enter();
+
+	/* does the scheduler think "t" was blocked on cos_rcv? */
+	sl_thd_sched_unblock_no_cs(t);
```

Comment:
Every time a thread blocks/yields, we check the status of `sched_blocked` to reset it because it could have run before the scheduler processes it's events. 
I'll try and rephrase the comments.